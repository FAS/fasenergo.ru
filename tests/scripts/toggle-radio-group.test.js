/**
 * @jest-environment jsdom
 */

/* eslint-env jest */
/* global GROUP, INPUT_1, INPUT_2 */

describe('ToggleSorters click listner', () => {
  it('should toggle inputs ordering', () => {
    document.body.innerHTML = `
      <form id='GROUP' class='js-toggle-sorter'>
        <label>First <input id='INPUT_1' name='sort' type='radio' data-order='asc' checked></label>
        <label>Second <input id='INPUT_2' name='sort' type='radio' data-order='asc'></label>
      </form>
    `

    require('../../source/scripts/plugins/toggle-radio-group')

    expect(GROUP.outerHTML).toMatchSnapshot()
    expect(INPUT_1.checked).toBe(true)
    expect(INPUT_2.checked).toBe(false)

    // We leave first input in cliecked state to ensure that
    // it will return to default state after INPUT_2 being clicked
    INPUT_1.click()

    expect(GROUP.outerHTML).toMatchSnapshot()
    expect(INPUT_1.checked).toBe(true)
    expect(INPUT_2.checked).toBe(false)

    INPUT_2.click()

    expect(GROUP.outerHTML).toMatchSnapshot()
    expect(INPUT_1.checked).toBe(false)
    expect(INPUT_2.checked).toBe(true)

    INPUT_2.click()

    expect(GROUP.outerHTML).toMatchSnapshot()
    expect(INPUT_1.checked).toBe(false)
    expect(INPUT_2.checked).toBe(true)

    INPUT_2.click()

    expect(GROUP.outerHTML).toMatchSnapshot()
    expect(INPUT_1.checked).toBe(false)
    expect(INPUT_2.checked).toBe(true)

    INPUT_1.click()

    expect(GROUP.outerHTML).toMatchSnapshot()
    expect(INPUT_1.checked).toBe(true)
    expect(INPUT_2.checked).toBe(false)
  })
})
