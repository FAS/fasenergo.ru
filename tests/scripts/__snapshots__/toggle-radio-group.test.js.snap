// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSorters click listner should toggle inputs ordering 1`] = `
"<form id=\\"GROUP\\" class=\\"js-toggle-sorter\\">
        <label>First <input id=\\"INPUT_1\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\" checked=\\"\\"></label>
        <label>Second <input id=\\"INPUT_2\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\"></label>
      </form>"
`;

exports[`ToggleSorters click listner should toggle inputs ordering 2`] = `
"<form id=\\"GROUP\\" class=\\"js-toggle-sorter\\">
        <label>First <input id=\\"INPUT_1\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"desc\\" checked=\\"\\"></label>
        <label>Second <input id=\\"INPUT_2\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\"></label>
      </form>"
`;

exports[`ToggleSorters click listner should toggle inputs ordering 3`] = `
"<form id=\\"GROUP\\" class=\\"js-toggle-sorter\\">
        <label>First <input id=\\"INPUT_1\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\" checked=\\"\\"></label>
        <label>Second <input id=\\"INPUT_2\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\"></label>
      </form>"
`;

exports[`ToggleSorters click listner should toggle inputs ordering 4`] = `
"<form id=\\"GROUP\\" class=\\"js-toggle-sorter\\">
        <label>First <input id=\\"INPUT_1\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\" checked=\\"\\"></label>
        <label>Second <input id=\\"INPUT_2\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"desc\\"></label>
      </form>"
`;

exports[`ToggleSorters click listner should toggle inputs ordering 5`] = `
"<form id=\\"GROUP\\" class=\\"js-toggle-sorter\\">
        <label>First <input id=\\"INPUT_1\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\" checked=\\"\\"></label>
        <label>Second <input id=\\"INPUT_2\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\"></label>
      </form>"
`;

exports[`ToggleSorters click listner should toggle inputs ordering 6`] = `
"<form id=\\"GROUP\\" class=\\"js-toggle-sorter\\">
        <label>First <input id=\\"INPUT_1\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\" checked=\\"\\"></label>
        <label>Second <input id=\\"INPUT_2\\" name=\\"sort\\" type=\\"radio\\" data-order=\\"asc\\"></label>
      </form>"
`;
